src/driver/driver_api.cc

```diff
@@ -47,6 +47,7 @@ TVM_REGISTER_PASS_CONFIG_OPTION("tir.disable_vectorize", Bool);
 TVM_REGISTER_PASS_CONFIG_OPTION("tir.is_entry_func", Bool);
 TVM_REGISTER_PASS_CONFIG_OPTION("tir.add_lower_pass", Array<Array<ObjectRef>>);
 TVM_REGISTER_PASS_CONFIG_OPTION("tir.debug_keep_trivial_loop", Bool);
+TVM_REGISTER_PASS_CONFIG_OPTION("tir.max_stack_alloca", Integer);
```

Comment:
Should we move this under namespace of the pass lower_tvm_builtin (e.g.  tir.lower_tvm_builtin.max_stack_alloca) ? and move it from driver_api into the pass implementation ?

Reply:
I looked into https://github.com/apache/tvm/blob/main/tests/python/unittest/test_tir_transform_lower_tvm_builtin.py
I am not quite sure how I would test that tir.max_stack_alloca works as expected. I would need to create an IRModule with AllocateNodes and check whether they are or are not in there after tranformations, depending on its value, right?
I do not really know how to do that unfortunately

Reply:
Hi @SebastianBoblestETAS ,

I guess you are anwsering the the question, I have asked in the title.
Let me cite an similar example : 

https://github.com/apache/tvm/blob/6720d3593d4dac6015418d4b7e9ad875bbf0b0a2/tests/python/unittest/test_tir_transform_inject_double_buffer.py#L56-L60

We could use a similiar tvm.tir.stmt_functor.post_order_visit that counts the remaining Allocate nodes.

For this thread,

What do you think about moving the PassConfig option to be part of configuration of lower_tvm_builtin pass rather than here in the driver_api ?  (because this configuration now essential becomes a configuration of the pass)

Reply:
Hi, sorry that this took me so long. I moved the option to lover_tvm_builtin.cc as proposed. I also created a test for this but it fails with 
Check failed: (device_type_.defined()) is false: Unknown device type in current IR

How can I specify a device?

I did not yet push this to this branch here, but the changes I made are here:
https://github.com/SebastianBoblestETAS/tvm/compare/makeMaxStackAllocaConfigurable...SebastianBoblestETAS:makeMaxStackAllocaConfigurableFix

Reply:
Hi @SebastianBoblestETAS ,

that information is usually inserted by Make(Un)PackedAPI transform and required by lower_builtin pass. Please take a look here : https://github.com/apache/tvm/blob/b365586159cd76a7fcd12d7915d2b59cc59e7105/tests/python/unittest/test_tir_transform_lower_tvm_builtin.py#L57-L68